### DISCUSSION TITLE
Any way to force base branch name inclusion in branch names?

### DISCUSSION THREAD

**Discussion by justinp** [renovatebot/renovate#22632] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.109.0

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'm trying to get renovate working with my CI system and build scripts. Our artifact versions are derived from the git branch names with certain patterns, so I need the branch names to be in a consistent format. There's a little more info in [another question I've asked](https://github.com/renovatebot/renovate/discussions/22589).

Well, I've got another path to making this work by teaching our build scripts how to interpret the default renovate branch names instead of modifying them. I have a PoC of that, but it requires the base branch name to be a part of the update branch name. This seems to be the case if and only if there are multiple base branches detected in the run. When there's only one base branch, the base branch name doesn't seem to be embedded there.

For example, when I have one baseBranch (using pattern `develop/.*`), I get:
```
 INFO: PR created (..., baseBranch=develop/10.0.0, branch=renovate/continuation-0.x)
```

When I add another matching base branches, I get:
```
 INFO: PR updated (..., baseBranch=develop/10.0.0, branch=renovate/develop/10.0.0-continuation-0.x)
...
 INFO: PR updated (..., baseBranch=develop/10.1.2, branch=renovate/develop/10.1.2-continuation-0.x)
```

In the first case, I want it to create a branch `renovate/develop/10.0.0-continuation-0.x` for base branch `develop/10.0.0` _even if_ there are no other base branches so that the name always consistently contains the name of the base branch.

### What you tried so far.

I haven't been able to find anything in the documentation to indicate that this is possible.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Try configuring `"additionalBranchPrefix": "{{baseBranch}}"`

**Comment by justinp** [comment]
Thanks. That works for one base branch. Unfortunately, that doubles up the base branch name where I have two matching branches:

```
 INFO: PR updated (..., baseBranch=develop/10.0.0, branch=renovate/develop/10.0.0-develop/10.0.0continuation-0.x)
...
 INFO: PR updated (..., baseBranch=develop/10.1.2, branch=renovate/develop/10.1.2-develop/10.1.2continuation-0.x)
```

I'd like it to be consistent no matter how many matching base branches there are.

**Reply by rarkins** [reply]
So only configure it on repos with one base branch

**Reply by justinp** [reply]
That's not ideal for us because we create and delete branches during development and I don't want to have to modify the renovate configuration every time someone does that.

**Reply by rarkins** [reply]
What does your base branches config look like?

**Reply by justinp** [reply]
```
  "baseBranches": ["/^develop\/.*/"],
```

**Reply by rarkins** [reply]
In this case we'd need to expose hasBaseBranches to templating too

**Reply by justinp** [reply]
Yeah, something like that is what I was looking for. I'm guessing from your response, there's no other way you can think of to do it currently?

Thanks for your help!

**Reply by rarkins** [reply]
That would be the only way. PR welcome to add it to available template fields
